From 4214fa420cbc506db2ac8d5280f1f06d1e75f262 Mon Sep 17 00:00:00 2001
From: Sergej Jaskiewicz <jaskiewiczs@icloud.com>
Date: Fri, 10 Jul 2020 20:35:24 +0300
Subject: [PATCH] Revert "Revert "[compiler-rt] [test] Use the parent process
 env as base env in tests""

This reverts commit 979c5023d3f0656cf51bd645936f52acd62b0333.

The underlying issue has been fixed in https://reviews.llvm.org/D83719.

(cherry picked from commit faa7e306e41ba250b1bf74511a6b0b15a944fb93)
---
 compiler-rt/test/lit.common.cfg.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compiler-rt/test/lit.common.cfg.py b/compiler-rt/test/lit.common.cfg.py
index 9d0c214bd9a..32a602bfb31 100644
--- a/compiler-rt/test/lit.common.cfg.py
+++ b/compiler-rt/test/lit.common.cfg.py
@@ -70,6 +70,8 @@ if config.android:
     # to link. In r19 and later we just use the default which is libc++.
     config.cxx_mode_flags.append('-stdlib=libstdc++')
 
+config.environment = dict(os.environ)
+
 # Clear some environment variables that might affect Clang.
 possibly_dangerous_env_vars = ['ASAN_OPTIONS', 'DFSAN_OPTIONS', 'LSAN_OPTIONS',
                                'MSAN_OPTIONS', 'UBSAN_OPTIONS',
-- 
2.28.0.236.gb10cc79966-goog

